iohub/ngff.py

```diff
@@ -1564,6 +1566,61 @@ def positions(self) -> Generator[tuple[str, Position], None, None]:
         for _, well in self.wells():
             for _, position in well.positions():
                 yield position.zgroup.path, position
+    
+    def rename_wells(self, well_names_csv: str):
+        """Modifies well names (and relevant zattrs) according to old and new paths specified in csv file. 
+
+        CSV input should be in the format:
+
+        oldrow/oldcolumn, newrow/newcolumn
+        
+        For example:
+
+        0/0, 0/A
+        0/1, 0/B
+        0/2, 0/C
+
+        rename_wells will adjust well paths from 0/0 --> 0/A, 0/1  -> 0/B
+
+        """
+        
+        names = []
+
+        with open(well_names_csv, mode='r', newline='', encoding='utf-8') as csvfile:
+            csvreader = csv.reader(csvfile)
+            for row in csvreader:
+                names.append([row[0], row[1]]) 
+
+        names = np.array(names)  ### names is a 2D array of [old_well_path, new_well_path
+
+        modified_wells = []
+        modified_columns = []
+        modified_rows = []
+
+        for old_well_path, new_well_path in names:
+            old_row, old_column = old_well_path.split('/')
+            new_row, new_column = new_well_path.split('/')
+
+            for well in self.metadata.wells:
+
```

Comment:
Double check that you have `black` installed. I think it will squash this whitespace, and I think that one's of the reasons CI isn't passing.